internal/api/v1/queryparams/query_parser.go

```diff
@@ -0,0 +1,104 @@
+package queryparams
+
+import (
+	"fmt"
+	"reflect"
+	"strings"
+
+	goclienterr "github.com/bitcoin-sv/spv-wallet-go-client/errors"
+	"github.com/bitcoin-sv/spv-wallet-go-client/queries"
+	"github.com/bitcoin-sv/spv-wallet/models/filter"
+)
+
+type QueryParser[F queries.QueryFilters] struct {
+	query *queries.Query[F]
+}
+
+func (q *QueryParser[F]) isNestedFilter(t reflect.Type) bool {
+	filters := []reflect.Type{
+		reflect.TypeOf(filter.UtxoFilter{}),
+		reflect.TypeOf(filter.TransactionFilter{}),
+		reflect.TypeOf(filter.PaymailFilter{}),
+		reflect.TypeOf(filter.ContactFilter{}),
+		reflect.TypeOf(filter.AccessKeyFilter{}),
+		reflect.TypeOf(filter.XpubFilter{}),
+		reflect.TypeOf(filter.ModelFilter{}),
+	}
+
+	for _, f := range filters {
+		if t == f {
+			return true
+		}
+	}
+	return false
```

Comment:
I like the new idea, but have a small suggestion. Let me know what do you think about something like this. This way we don't have to add new filters each time here, but rather we check if any filter has ModelFilter inside it and it it does we could assume it should return true here?

```suggestion
	func (q *QueryParser[F]) checkForFilter(t reflect.Type) bool {
	// Check if the type directly matches ModelFilter
	if t == reflect.TypeOf(filter.ModelFilter{}) {
		return true
	}

	// If the input is a struct, check its fields for ModelFilter
	if t.Kind() == reflect.Struct {
		for i := 0; i < t.NumField(); i++ {
			field := t.Field(i)
			if field.Type == reflect.TypeOf(filter.ModelFilter{}) {
				return true
			}
		}
	}

	return false
}
```

Reply:
I think it would always work if we add any new filter that embeds ModelFilter and this way there it works out of the box

Reply:
I was wondering if this approach would work for structures without nested model filters or for custom structures with JSON tags. However, it works as expected! 😊 Great job—I'll go ahead and apply your suggestion.